default/stringtables/en.txt

```diff
@@ -11128,6 +11128,8 @@ Energy Compressor
 BLD_SHIPYARD_ENRG_COMP_DESC
 '''This is an upgrade to the [[buildingtype BLD_SHIPYARD_BASE]] and is required for the formation of all energy hulls and all further energy hull shipyard upgrades.
 
+Requiring a high density star to siphon energy from, it can only be built at a system with a [[STAR_BLUE]], [[STAR_WHITE]] or [[STAR_BLACK]] star type.
```

Comment:
I agree "high density" is not quite right.

We could try "high gravity", "high spatial curvature",  "high space-time curvature" or "high mass" as other perhaps better descriptions of the imaginary technology.

"high energy"  works too.

Or ask your friend, a brown dwarf expert is qualified to write more correct future tech description of stellar mass/energy harvesting.